src/sage/rings/polynomial/laurent_polynomial.pyx

```diff
@@ -1318,22 +1318,59 @@ cdef class LaurentPolynomial_univariate(LaurentPolynomial):
             return ~self
         raise ArithmeticError("element is not a unit")
 
+    @coerce_binop
     def xgcd(self, other):
-        """
+        r"""
         Extended `gcd` for univariate Laurent polynomial rings over a field.
 
+        OUTPUT:
+
+        A triple ``(g, p, q)`` such that ``g`` is the :meth:`gcd` of
+        ``self`` (`= a`) and ``other`` (`= b`), and ``p`` and ``q`` are
+        cofactors satisfying the Bezout identity
+
+        .. MATH::
+
+            g = p \cdot a + q \cdot b.
+
         EXAMPLES::
 
             sage: S.<t> = LaurentPolynomialRing(QQ)
-            sage: (t^-2 + 1).xgcd(t^-3 + 1)
-            (1, 1/2*t^2 - 1/2*t^3 - 1/2*t^4, 1/2*t^3 + 1/2*t^4)
+            sage: a = t^-2 + 1
+            sage: b = t^-3 + 1
+            sage: g, p, q = a.xgcd(b); (g, p, q)
+            (t^-3, 1/2*t^-1 - 1/2 - 1/2*t, 1/2 + 1/2*t)
+            sage: g == p * a + q * b
+            True
+            sage: g == a.gcd(b)
+            True
+            sage: t.xgcd(t)
+            (t, 0, 1)
+            sage: t.xgcd(5)
+            (1, 0, 1/5)
         """
-        R = self.parent()
-        S = R.polynomial_ring()
-        f, df = self.monomial_reduction()
-        g, dg = other.monomial_reduction()
-        h, p, q = f.xgcd(g)
-        return R(h), p / df, q / dg
+        cdef LaurentPolynomial_univariate elt = other
+        cdef LaurentPolynomial_univariate ret_gcd, ret_p, ret_q
+        cdef long n = min(self.__n, elt.__n)
+
+        h, p, q = self.__u.xgcd(elt.__u)
+
```

Comment:
I think mathematically it would be better to take out the `x` factors of `h` and change accordingly `p` and `q` to have Bézout identity. It would imply to change `gcd`.